computer-vision/robot_tracking.cpp

```diff
@@ -0,0 +1,171 @@
+#include <opencv2/highgui.hpp>
+#include <opencv2/aruco.hpp>
+#include <opencv2/calib3d.hpp>
+
+#include <iostream>
+
+#include "protobuf/protocol.pb.h"
+#include "protobuf/computer_vision.pb.h"
+#include "error/error.h"
+#include "logging.h"
+#include "udp_protocol.h"
+
+#define ARUCO_CENTER 42
+#define ARUCO_WEATHERVANE 17
+
+struct RobotTrackingSettings {
+    int image_width;
+    int image_height;
+    std::string destination_ip;
+    int destination_port;
+    cv::Mat camera_matrix;
+    cv::Mat distortion_coefficients;
+
+    void load(const std::string& file)
+    {
+        int flags = cv::FileStorage::READ; // cv::FileStorage::FORMAT_YAML
+        cv::FileStorage fs = cv::FileStorage(file, flags);
+
+        cv::read(fs["image_width"], image_width, -1);
+        cv::read(fs["image_height"], image_height, -1);
+        cv::read(fs["camera_matrix"], camera_matrix);
+        cv::read(fs["distortion_coefficients"], distortion_coefficients);
+
+        destination_ip = (std::string)fs["destination_ip"];
+        destination_port = (int)fs["destination_port"];
+    }
+};
+
+enum WeathiervaneDirection {
+    NORTH = 0,
+    SOUTH = 1
+};
+
+WeathiervaneDirection weathervane_direction(cv::Vec3d rvec, cv::Vec3d tvec)
```

Comment:
```suggestion
WeathervaneDirection weathervane_direction(cv::Vec3d rvec, cv::Vec3d tvec)
```
typo?

Reply:
Fixed by removing the intermediate enum in favor of the protobuf-declared one.